dynatrace/build.gradle

```diff
@@ -26,12 +26,15 @@ repositories {
     mavenCentral()
 }
 
-def otelVersion = '1.14.0'
-
 dependencies {
-    implementation platform("io.opentelemetry:opentelemetry-bom:${otelVersion}")
-    implementation('io.opentelemetry:opentelemetry-sdk')
-    implementation('io.opentelemetry:opentelemetry-sdk-metrics')
+    compileOnly platform("io.opentelemetry:opentelemetry-bom") {
+        version {
+            strictly '[1.14.0, 2.0['
+            prefer '1.20.1'
+        }
+    }
+    compileOnly('io.opentelemetry:opentelemetry-sdk')
+    compileOnly('io.opentelemetry:opentelemetry-sdk-metrics')
```

Comment:
This is what I was thinking: https://github.com/dynatrace-oss/opentelemetry-metric-java/compare/tct-test?expand=1

AFAIU, the version and strictly that you defined above means: when you build the lib, build it against any OTel SDK version between 1.14 and 2.0, but prefer 1.20.1. So, it will probably always build it against 1.20.1. 

It will compile against that version, but with compileOnly it won't include the OTel SDK in the JAR, so you will have to provide your own OTel SDK when using this lib. I don't think there is a way to specify which version range the consuming package can use.